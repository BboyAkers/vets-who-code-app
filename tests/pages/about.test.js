import React from 'react';
import About from '../../src/pages/about';
import jQuery from '../../static/vendor/jquery/dist/jquery';

describe('<About />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<About />);
  });

  test.skip('should invoke play method on click', () => {
    wrapper.find('.play-button').simulate('click');
    const spy = jest.spyOn(wrapper.instance(), 'play');
    // global.HTMLMediaElement.prototype.play = () => ({});
    // wrapper.instance().forceUpdate();
    expect(spy).toHaveBeenCalled();
  });

  test('should invoke end method when video completes', () => {
    const spy = jest.spyOn(wrapper.instance(), 'end');
    window.HTMLMediaElement.prototype.end = wrapper.instance().end();
    expect(spy).toHaveBeenCalled();
  });

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
