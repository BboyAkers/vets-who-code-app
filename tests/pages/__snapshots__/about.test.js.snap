// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> should render correctly 1`] = `
<TemplateWrapper>
  <header
    className="inner-header overlay grey text-center slim-bg "
    style={
      Object {
        "backgroundImage": "url(image-mock)",
        "backgroundPositionY": "bottom",
      }
    }
  >
    <div
      className="overlay-01"
    />
    <div
      className="container"
    >
      <h2
        className="text-center text-uppercase"
      >
        About
      </h2>
      <div
        className="breadcrumb"
      >
        <Link
          to="/"
        >
          Home
        </Link>
        <span>
          /
        </span>
        <Link
          className="page-active"
          to="/about"
        >
          About
        </Link>
      </div>
    </div>
  </header>
  <section
    className="small-top-pad section bg-default"
    id="about"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <h3
            className="story-title"
          >
            Our Success story
          </h3>
        </div>
        <div
          className="col-md-5"
          style={
            Object {
              "marginBottom": 20,
            }
          }
        >
          <div
            className="success-video"
            style={
              Object {
                "position": "relative",
              }
            }
          >
            <video
              className="img-responsive"
              muted={true}
              type="video/mp4"
            >
              <source
                src="image-mock"
              />
            </video>
            <a
              className="play-button"
              onClick={[Function]}
            >
              <i
                className="fa fa-play-circle-o"
                style={
                  Object {
                    "color": "#0e203e",
                    "cursor": "pointer",
                    "fontSize": 40,
                    "left": "50%",
                    "position": "absolute",
                    "top": "50%",
                    "transform": "translate(-50%, -50%)",
                  }
                }
              />
            </a>
          </div>
          <div
            className="vwc-animated-gif"
            style={
              Object {
                "backgroundImage": "url(image-mock)",
                "backgroundPosition": "center center",
                "backgroundSize": "cover",
              }
            }
          />
        </div>
        <div
          className="col-md-7"
        >
          <div
            className="success-story"
          >
            <p
              className="story"
            >
              Launched in 2014, Vets Who Code is a non-profit dedicated to filling the nations technical skills gap with America’s best. We achieve this by using technology to connect and train veterans remotely in web development in order to close the digital talent gap and ease career transition for military veterans and to give military spouses skills to provide stability as they move to support their families. We believe that those who serve in uniform can be the digital economy’s most productive and innovative . Vets Who Code prepares them to enter the civilian work force with tangible skills for new careers.
            </p>
            <p
              className="story-last"
            >
              Vets Who Code is a reintegration solution for veterans that believes in the principle of "To Teach a Man To Fish" to better prepare early stage transitioning veterans for returning to the workforce. Many veterans are not homeless, but are un- and under-employed. Some have a deep desire to be independent business owners. Their skill sets are desperately needed to enhance local economies and to drive innovation. Vets Who Code serves as a launch pad for our veterans and military spouses.
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section
    className="section bg-default"
    id="team"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <h3
            className="meet-us-heading"
          >
            Our Team
          </h3>
          <div
            className="row"
          >
            <div
              className="col-md-4 col-sm-6"
            >
              <div
                className="team-wiget clearfix"
              >
                <img
                  alt="Jerome Hardaway Head Geek"
                  className="img-responsive"
                  src="image-mock"
                />
                <div
                  className="meet-social"
                >
                  <a
                    href="https://twitter.com/JeromeHardaway"
                  >
                    <i
                      className="fa fa-twitter"
                    />
                  </a>
                  <a
                    href="https://github.com/jeromehardaway"
                  >
                    <i
                      className="fa fa-github"
                    />
                  </a>
                  <a
                    href="https://www.linkedin.com/in/jeromehardaway/"
                  >
                    <i
                      className="fa fa-linkedin"
                    />
                  </a>
                </div>
                <div
                  className="personal-info"
                >
                  <p
                    className="title"
                  >
                    Jerome Hardaway
                  </p>
                  <p
                    className="designation"
                  >
                    Head Geek
                  </p>
                </div>
              </div>
            </div>
            <div
              className="col-md-4 col-sm-6"
            >
              <div
                className="team-wiget clearfix"
              >
                <img
                  alt="Andrew Lebowitz Brand Geek"
                  className="img-responsive"
                  src="image-mock"
                />
                <div
                  className="meet-social"
                >
                  <a
                    href="https://twitter.com/witzcreative"
                  >
                    <i
                      className="fa fa-twitter"
                    />
                  </a>
                  <a
                    href="https://www.behance.net/WitzCreative"
                  >
                    <i
                      className="fa fa-behance"
                    />
                  </a>
                  <a
                    href="https://www.linkedin.com/in/witzcreative"
                  >
                    <i
                      className="fa fa-linkedin"
                    />
                  </a>
                </div>
                <div
                  className="personal-info"
                >
                  <p
                    className="title"
                  >
                    Andrew Lebowitz
                  </p>
                  <p
                    className="designation"
                  >
                    Brand Geek
                  </p>
                </div>
              </div>
            </div>
            <div
              className="col-md-4 col-sm-6"
            >
              <div
                className="team-wiget clearfix"
              >
                <img
                  alt="Noel Sagaille Devops Geek"
                  className="img-responsive"
                  src="image-mock"
                />
                <div
                  className="meet-social"
                >
                  <a
                    href="https://twitter.com/MrBernnz"
                  >
                    <i
                      className="fa fa-twitter"
                    />
                  </a>
                  <a
                    href="https://github.com/mrbernnz"
                  >
                    <i
                      className="fa fa-github"
                    />
                  </a>
                  <a
                    href="https://www.linkedin.com/in/mrbernnz"
                  >
                    <i
                      className="fa fa-linkedin"
                    />
                  </a>
                </div>
                <div
                  className="personal-info"
                >
                  <p
                    className="title"
                  >
                    Noel Sagaille
                  </p>
                  <p
                    className="designation"
                  >
                    Devops Geek
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</TemplateWrapper>
`;
